### PR TITLE
[issue/16] - added notifications part

### LINKED ISSUE #16: Add notifications and client section for app

### PR DESCRIPTION
**Pull request by diehmlukas**
close #16

### CODE REVIEW

File: report/04_implementation/01_flutter_application.md [hunk 1]
```diff
@@ -25,6 +25,16 @@ logic.
 
 ### Notifications
 
+The app is able to send reminder notifications to the user. This can be set by 
+the researcher, how often and on which times this should happen. In this way, 
+the study participant can be specifically reminded to use the app. The setting
```
Comment (mac641) [inline report/04_implementation/01_flutter_application.md:30]:
```suggestion
the study participant can be specifically reminded to use the app. The settings 
```

File: report/04_implementation/01_flutter_application.md [hunk 2]
```diff
@@ -25,6 +25,16 @@ logic.
 
 ### Notifications
 
+The app is able to send reminder notifications to the user. This can be set by 
+the researcher, how often and on which times this should happen. In this way, 
+the study participant can be specifically reminded to use the app. The setting 
+options can be found in the config, here both the desired notification times in 
+UTC and the notification text can be set. The heading of each notification is
```
Comment (mac641) [inline report/04_implementation/01_flutter_application.md:32]:
```suggestion
option can be found in the config, where both the desired notification times 
formatted as UTC and the notification text can be set. The heading of each notification is 
```

**Review by mac641 — CHANGES_REQUESTED** [review]
Just some small improvement suggestions ;)

**Review by AnnaGsell — DISMISSED** [review]

File: report/04_implementation/01_flutter_application.md [hunk 3]
```diff
@@ -25,6 +25,19 @@ logic.
 
 ### Notifications
 
+The app is able to send reminder notifications to the user.
+The researcher can configure specific times, when notifications will be triggered. 
+In this way, the study participant can be specifically reminded to use the app. 
+The settings option can be found in the config, where both the desired notification 
+times formatted as UTC and the notification text can be set. The heading of each 
+notification is the set study name. Further methods have been implemented which
```
Comment (carina-str) [inline report/04_implementation/01_flutter_application.md:33]:
```suggestion
notification is the study name, which is set in the configuration of the app. Further methods have been implemented which 
```

**Review by carina-str — APPROVED** [review]

File: report/04_implementation/01_flutter_application.md [hunk 4]
```diff
@@ -25,6 +25,20 @@ logic.
...
+The settings option can be found in the config, where both the desired notification 
+times formatted as UTC and the notification text can be set. The heading of each 
+notification is the study name, which is set in the configuration of the app. 
+Further methods have been implemented which can be used to send notifications 
+to the user at intervals or after certain events. These methods were only added 
+to make the app easier to expand and do not play a role in the current version. 
+One way to implement the notifications would have been to use firebase. 
+The team decided against this option in order to strengthen the stand alone app 
+concept. Instead, the flutter_local_notifications package was used, which makes 
+it possible to make the app more independent of a server.
```
Comment (treilik) [inline report/04_implementation/01_flutter_application.md:40]:
```suggestion
concept and the privacy and locality of the users data. Instead, the flutter_local_notifications
package was used, which makes it possible to make the app more independent of a server.
```

**Review by treilik — APPROVED** [review]

**Review by mac641 — APPROVED** [review]

**Pull request merged**